praw/models/reddit/mixins/__init__.py

```diff
@@ -134,10 +139,52 @@ def remove(self, spam=False):
            # remove a submission
            submission = reddit.submission(id='5or86n')
            submission.mod.remove()
+           # remove a submission with a removal reason
+           reason = reddit.subreddit.reasons["110ni21zo23ql"]
+           submission = reddit.submission(id="5or86n")
+           submission.mod.remove(reason_id=reason.id)
 
         """
         data = {"id": self.thing.fullname, "spam": bool(spam)}
         self.thing._reddit.post(API_PATH["remove"], data=data)
+        if any([reason_id, mod_note]):
+            self.add_removal_reason(reason_id, mod_note)
+
+    def add_removal_reason(self, reason_id=None, mod_note=""):
```

Comment:
While this is a separate API call, I think this should be a `_private` method. I see little reason why anyone would want to call this individually rather than in combination with remove. Thoughts?

The one benefit I do see of having this public, is it appears we could call this method with multiple items if we added an `other_items` parameter.

Reply:
I made `add_removal_reason` private (by just putting a single _ in the front - this is correct, no?)

I see that other private methods do not get referenced in docstrings so I removed the references to `:meth:_add_removal_reason` as sphinx was throwing an error. Instead I just mention that, if `reason_id` and/or `mod_note` are provided, a second API call with removal reason will be added to the item.

As for `other_items` I know that Reddit's API for adding a removal reason takes an `item_ids` but I haven't tested using it with more than one item. I believe that making it private as you suggest is the better move for now.